sdk/keyvault/azure-security-keyvault/azure/security/keyvault/secrets/README.md

```diff
@@ -0,0 +1,231 @@
+
+ * Create a service principal and configure its access to Azure resources:
+    ```Bash
+    az ad sp create-for-rbac -n <your-application-name> --skip-assignment
+    ```
+    Output:
+    ```json
+    {
+        "appId": "fa7c8686-c4f5-4639-ab88-497ee122bad3",
+        "displayName": "myspn",
+        "name": "http://myspn",
+        "password": "30e5c628-a322-3497-8429-52f2f98788c7",
+        "tenant": "72f988bf-86f1-41fg-91ab-2d7cd011db47"
+    }
+    ```
+* Use the above returned credentials information to set **AZURE_CLIENT_ID**(appId), **AZURE_CLIENT_SECRET**(password) and **AZURE_TENANT_ID**(tenant) environment variables.
+  ```Bash
+    export AZURE_CLIENT_ID="fa7c8686-c4f5-4639-ab88-497ee122bad3"
+    export AZURE_CLIENT_SECRET="30e5c628-a322-3497-8429-52f2f98788c7"
+    export AZURE_TENANT_ID="72f988bf-86f1-41fg-91ab-2d7cd011db47"
+  ```
+
+* Grant the above mentioned application authorization to perform secret operations on the keyvault:
+    ```Bash
+    az keyvault set-policy --name <your-key-vault-name> --spn $AZURE_CLIENT_ID --secret-permissions backup delete get list set
+    ```
+    > --secret-permissions:
+    > Accepted values: backup, delete, get, list, purge, recover, restore, set
+
+#### Create Secret client
+Once you've populated the **AZURE_CLIENT_ID**, **AZURE_CLIENT_SECRET** and **AZURE_TENANT_ID** environment variables, you can create the [SecretClient](https://github.com/Azure/azure-sdk-for-python/blob/master/sdk/keyvault/azure-security-keyvault/azure/security/keyvault/secrets/_client.py):
+
+```python
+    from azure.identity import DefaultAzureCredential
+    from azure.security.keyvault import SecretClient
+
+    credential = AsyncDefaultAzureCredential()
+
+    # Create a new secret client using a client secret credential
+    secret_client = SecretClient(vault_url=<your-vault-url>, credential=credential)
+```
+## Key concepts
+### Secret
+  A secret is the fundamental resource within Azure KeyVault. From a developer's perspective, Key Vault APIs accept and return secret values as strings. In addition to the secret data, the following attributes may be specified:
+* expires: Identifies the expiration time on or after which the secret data should not be retrieved.
+* not_before: Identifies the time after which the secret will be active.
+* enabled: Specifies whether the secret data can be retrieved.
+* created: Indicates when this version of the secret was created.(read-only)
+* updated: Indicates when this version of the secret was updated.(read-only)
+
+### Secret Client:
+The Secret client performs the interactions with the Azure Key Vault service for getting, setting, updating,deleting, and listing secrets and its versions. An asynchronous and synchronous, SecretClient, client exists in the SDK allowing for selection of a client based on an application's use case. Once you've initialized a SecretClient, you can interact with the primary resource types in Key Vault.
```

Comment:
I think `SecretClient` and its purpose belong at the top--the purpose of this package is to provide `SecretClient`, the API for secrets.